src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -317,7 +317,7 @@ private string GetProcessesToIgnoreKey(long hostHandshake, long clientHandshake,
         private void ValidateRemotePipeSecurityOnWindows(NamedPipeClientStream nodeStream)
         {
             SecurityIdentifier identifier = WindowsIdentity.GetCurrent().Owner;
-#if FEATURE_PIPE_SECURITY
+#if !FEATURE_PIPE_SECURITY
```

Comment:
This code isn't currently used on .NET Core.  Initially I ported the /m support using named pipes, and as part of that added the local copy of PipeSecurity and factored this code out into a separate method.  Subsequently, I switched to using anonymous pipes in .NET Core.  However, if we want to add support for node reuse to .NET Core at some point, then we would probably need to go back to using named pipes.

In any case, I am still confused and I think this may be the wrong change.  `FEATURE_PIPE_SECURITY` was supposed to mean "the Framework includes PipeSecurity", ie the opposite of what `FEATURE_USE_LOCAL_COPY_PIPE_SECURITY` would be.  When PipeSecurity is included in the framework, then you can call GetAccessControl on the stream.  If it's not included, then you have to call the constructor for PipeSecurity (and it will be the local copy of the code).

What was the problem with the Mono build?  Does Mono include PipeSecurity?